internal/armada/server/submit.go

```diff
@@ -689,35 +695,41 @@ func (server *SubmitServer) checkReprioritizePerms(ctx context.Context, jobs []*
 	return nil
 }
 
-func (server *SubmitServer) getQueueOrCreate(ctx context.Context, queueName string) (queue.Queue, error) {
+func (server *SubmitServer) getQueueOrCreate(ctx context.Context, queueName string) (*queue.Queue, error) {
 	q, e := server.queueRepository.GetQueue(queueName)
 	if e == nil {
-		return q, nil
+		return &q, nil
 	}
 	var expected *repository.ErrQueueNotFound
 
 	if errors.As(e, &expected) {
-		if !server.queueManagementConfig.AutoCreateQueues || !server.permissions.UserHasPermission(ctx, permissions.SubmitAnyJobs) {
-			return queue.Queue{}, status.Errorf(codes.NotFound, "Queue %q not found", queueName)
+
+		if !server.queueManagementConfig.AutoCreateQueues {
+			return nil, status.Errorf(codes.Aborted, "Queue %s not found; refusing to make it automatically (server setting autoCreateQueues is false)", queueName)
+		}
+		if !server.permissions.UserHasPermission(ctx, permissions.SubmitAnyJobs) {
+			return nil, status.Errorf(codes.PermissionDenied, "Queue %s not found; won't create because user lacks SubmitAnyJobs permission", queueName)
 		}
 
 		principal := authorization.GetPrincipal(ctx)
-
+		groupNames := principal.GetGroupNames()
+		everyone_index := slices.Index(groupNames, "everyone")
+		groupNames = append(groupNames[:everyone_index], groupNames[everyone_index+1:]...)
```

Comment:
Several nit picks here:
 - Make make remove string from slice a separate function so it can be tested (there are some util libraries in common
  - Use the "everyone" constant https://github.com/G-Research/armada/blob/master/internal/common/auth/authorization/common.go#L23
   - Have a comment saying why we want to remove this